dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java

```diff
@@ -128,9 +124,40 @@ public static SourceCodeRepoInterface createSourceCodeRepo(String gitUrl, String
      * @param url
      * @return a map with keys: Source, Username, Repository
      */
-    public static Map<String, String> parseGitUrl(String url) {
-        Matcher m1 = GITHUB_REGEX_PATTERN_1.matcher(url);
-        Matcher m2 = GITHUB_REGEX_PATTERN_2.matcher(url);
+    public static Optional<Map<String, String>> parseGitUrl(String url) {
+        return parseGitUrl(url, Optional.empty());
+    }
+
+    /**
+     * Parse Git URL to retrieve source, username and repository name.
+     *
+     * @param url
+     * @param repositoryName fixed repository name to match in git URL
```

Comment:
I'd skimmed over this in the first review and without looking at the regex closely, based on the name I thought it was the repository, e.g., `dockstore-ui2` in `github.com/dockstore/dockstore-ui2`, when it's `github.com`. Even though we already use `Repository` as a key in the map the same way as you're using it, I'd still suggest some combination of:

1. Calling it host or gitHost
2. This comment listing the possible values (github.com, bitbucket.org, gitlab.org).
3. Or have an enum so it can't possibly be called with the wrong string value.

Reply:
Did 1 and 2 but not 3.